src/main/java/usi/si/seart/repository/criteria/Criteria.java

```diff
@@ -1,3 +1,15 @@
 package usi.si.seart.repository.criteria;
 
-public interface Criteria {}
+import usi.si.seart.model.GitRepo;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.Path;
+import javax.persistence.criteria.Predicate;
+import java.util.List;
+
+public interface Criteria {
+
+    default List<Predicate> expand(Path<GitRepo> path, CriteriaBuilder criteriaBuilder) throws UnsupportedOperationException {
+        throw new UnsupportedOperationException("Expand operation on Paths is not supported!");
+    }
```

Comment:
No need for the `default` method. All children of `Criteria` have to implement the method. Just change this definition to:
```java
	List<Predicate> expand(Path<GitRepo> path, CriteriaBuilder criteriaBuilder);
```
Also, `UnsupportedOperationException` is a subclass of `RuntimeException`, and as such does not need to be declared in the `throws` clause. If you want to signify that a method throws a runtime exception, do so in the JavaDoc.